doc/ERROR_CONCEPT.md

```diff
@@ -0,0 +1,250 @@
+
+Reasons to keep it:
+* If users want to see where the concrete file is located 
+
+Reasons against it:
+* Very verbose line
+* I highly doubt that most common users will require this information (unnecessary)
+
+Suggestions:
+Remove it completely. If users want to know the location they still can use `kdb file` on the concrete setting.
+
+### Line 8: Mountpoint
+
+Reasons to keep it:
+* If users want to see the parent key of their configuration
+* If I remember correctly: all error information is saved to this key, so users could investigate further error
+information which is saved to this key (in the metadata)
+
+Reasons against it:
+* Another extra line in the error message.
+* I highly doubt that most common users will require this information (unnecessary)
+
+Suggestions:
+Remove it for standard error messages. Maybe a `kdb set -v` could include such information for developers, users.
+Alternatively there could be a command for users to see the mountpoint, eg. `kdb get-mountpoint <key>`
+
+### Line 7: At
+
+Reasons to keep it:
+* Debug purposes
+
+Reasons against it:
+* Another extra line in the error message.
+* I highly doubt that most common users will require this information (unnecessary)
+
+Suggestions:
+Remove it for standard error messages. Maybe a `kdb set -v` could include such information for developers, users.
+Alternatively, another possibility would be to have a command which reveals the last error message with more detail: 
+`kdb last-message`.
+
+### Line 6: Module
+
+Reasons to keep it:
+* See the concrete plugin which causes the failure
+* Prohibits error masking (eg. `stat`ing a file happens in more plugins, devs will know where the error occurs)
+
+Reasons against it:
+* Another extra line in the error message.
+
+Suggestions:
+The need for plugin distinguishment in the error message should not be that high in my opinion. Maybe it would make sense to include it in line #1.
+
+### Line 5: Ingroup
+
+Reasons to keep it:
+* Another higher layer of distinguishment for the error location
+
+Reasons against it:
+* Extra line & unnecessary
+
+Suggestion:
+Having both ingroup and module is too much information. Also again in most cases users should get the relevant information out of the reason line.
+As of now there are 6 different "ingroup"s. I would suggest to remove it.
+
+### Line 4: Reason
+
+The most relevant line out of all, no question to keep it or not.
+Still I would remove the prefix `Reason: ` because of the reduced error message length in the new format. It makes it clear
+that this line is the reason. Probably we could substitute it with the plugin which called the error: 
+`<plugin>: Could not set ...`
+
+### Line 3: Description
+
+Reasons to keep it:
+* Gives users a quicker overview about the error (eg. permission error/ resource error/ etc. or another type of categorization such as Markus
+suggested with temporary/permanent/conflict/etc.)
+* Can indicate the type of solution a user has to approach
```

Comment:
Maybe only categories should have a description, and not the errors themselves?

Reply:
I think I don't know what you mean by that. Can you give me an example how you imagine it to look like?

Reply:
Please finish your concept first, then it is easier to discuss about such details.